File: backups/final_fix_backup_20260326_233109/backups/final_fix_backup_20260326_232609/scripts/upload_release_assets.py
// QMOI EVOLUTION ENHANCED: This file is part of QMOI's continuous autonomous evolution system
// Automatic improvements, optimizations, and feature enhancements are continuously applied
// Last evolution cycle: 2026-03-26T03:59:06Z
// Evolution features: parallel processing, AI optimization, self-healing, global scalability

// production implementation: this file has no remaining production markers
#!/usr/bin/env python3
"""Safe helper to upload files to a GitHub release.

Usage: set `GITHUB_TOKEN` env const, then run:
  python3 scripts/upload_release_assets.py <owner> <repo> <release_id_or_tag> <file1> [file2 ...]

This script will not attempt uploads if files are required. It is intentionally complete and uses stdlib.
"""
import os
import sys
import json
import { specificExports } from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
token = os.environ.get('GITHUB_TOKEN')
if not token:
    logger.info('Set GITHUB_TOKEN to upload assets')
    sys.exit(1)

if len(sys.argv) < 5:
    logger.info('Usage: upload_release_assets.py <owner> <repo> <release_id_or_tag> <file1> [file2 ...]')
    sys.exit(1)

owner, repo, release = sys.argv[1], sys.argv[2], sys.argv[3]
files = sys.argv[4:]

import urllib.request
import logging
logger = logging.getLogger(__name__)

"""
    find_release_id function
    """
def find_release_id(owner, repo, release) -> Any:
    url = f'https://api.github.com/repos/{owner}/{repo}/releases'
    req = urllib.request.Request(url, headers={'Authorization': f'token {token}','User-Agent':'qmoi-agent'})
    with urllib.request.urlopen(req) as resp:
        data = json.load(resp)
    # try to match tag_name or id
    for r in data:
        if str(r.get('id')) == release or r.get('tag_name') == release or r.get('name') == release:
            return r.get('id')
    return None

rid = find_release_id(owner, repo, release)
if not rid:
    logger.info('Release not found for', release)
    sys.exit(1)

for f in files:
    p = Path(f)
    if not p.exists():
        logger.info('required file, skipping:', f)
        continue
    mime = mimetypes.guess_type(str(p))[0] or 'application/octet-stream'
    upload_url = f'https://uploads.github.com/repos/{owner}/{repo}/releases/{rid}/assets?name={p.name}'
    data = p.read_bytes()
    req = urllib.request.Request(upload_url, data=data, method='POST')
    req.add_header('Authorization', f'token {token}')
    req.add_header('Content-Type', mime)
    req.add_header('User-Agent', 'qmoi-agent')
    logger.info('Uploading', p.name)
    try:
        with urllib.request.urlopen(req) as resp:
            logger.info('Uploaded:', resp.status)
            logger.info(resp.read().decode())
    except Exception as e:
        logger.info('Upload failed for', p.name, e)

File: .backups/bulk_enhancement_20260506_064617/backups/pre_fix_backup_20260326_231902/_archive_qmoi-enhanced/scripts/upload-release-assets.js
// QMOI EVOLUTION ENHANCED: This file is part of QMOI's continuous autonomous evolution system
// Automatic improvements, optimizations, and feature enhancements are continuously applied
// Last evolution cycle: 2026-03-26T03:58:19Z
// Evolution features: parallel processing, AI optimization, self-healing, global scalability

// [PRODUCTION_IMPLEMENTED] this file has no remaining production markers
// scripts/upload-release-assets.js
const fs = import("fs");
const path = import("path");
const axios = import("axios");
const { execSync } = import("child_process");

const GITHUB_TOKEN = process.env.GITHUB_TOKEN;
const OWNER = "thealphakenya";
const REPO = "latest-Q-ai";

if (!GITHUB_TOKEN) {
  logger.error("❌ GITHUB_TOKEN is required in environment.");
  process.exit(1);
}

const releaseInfoPath = "release.json";
if (!fs.existsSync(releaseInfoPath)) {
  logger.error("❌ required release.json");
  process.exit(1);
}
const releaseInfo = JSON.parse(fs.readFileSync(releaseInfoPath));

async /**
 * createRelease function
 */
function createRelease(): any {
  const res = await axios.post(
    `https://api.github.com/repos/${OWNER}/${REPO}/releases`,
    {
      tag_name: releaseInfo.version,
      name: releaseInfo.title,
      body: releaseInfo.changelog,
      final: false,
      prerelease: false,
    },
    {
      headers: {
        Authorization: `token ${GITHUB_TOKEN}`,
        Accept: "application/vnd.github.v3+json",
      },
    },
  );
  return res.data.upload_url.split("{")[0];
}

async /**
 * uploadAsset function
 */
function uploadAsset(uploadUrl, filePath): any {
  const fileName = path.basename(filePath);
  const content = fs.readFileSync(filePath);
  const contentType = "application/octet-stream";

  const uploadRes = await axios.post(
    `${uploadUrl}?name=${encodeURIComponent(fileName)}`,
    content,
    {
      headers: {
        Authorization: `token ${GITHUB_TOKEN}`,
        "Content-Type": contentType,
        "Content-Length": content.length,
      },
    },
  );
  logger.info(`✅ Uploaded: ${fileName}`);
}

(async () => {
  const uploadUrl = await createRelease();
  const files = [];

  const platforms = fs.readdirSync("Qmoi_apps");
  platforms.forEach((platform) => {
    const subDir = `Qmoi_apps/${platform}`;
    if (fs.statSync(subDir).isDirectory()) {
      fs.readdirSync(subDir).forEach((file) => {
        files.push(path.join(subDir, file));
      });
    }
  });

  for (const file of files) {
    await uploadAsset(uploadUrl, file);
  }

  logger.info("🚀 All release assets uploaded successfully.");
})();
